src/Core/CommandBus/ExecutedCommandRegistry.php

```diff
@@ -114,8 +115,8 @@ private function getTrace(): array
     {
         $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, self::BACKTRACE_LIMIT);
 
-        foreach ($trace as $step) {
-            if ($step['class'] === TacticianCommandBusAdapter::class
+        foreach ($trace as $key => $step) {
+            if (is_a($step['class'], MessengerCommandBus::class, true)
                 && $step['function'] === 'handle'
```

Comment:
Could you use the interface instead of the concrete class here please? Also could you reverse both conditions, I think it will be a bit more optimized to check for the function name first as `is_a` may require more resource than a simple string equality

```suggestion
            if ($step['function'] === 'handle'
                && is_a($step['class'], CommandBusInterface::class, true)
```

Reply:
Well seen, it will indeed be more optimized